app/models/katello/glue/candlepin/activation_key.rb

```diff
@@ -37,12 +37,15 @@ def get_key_pools
       end
 
       def import_pools
-        pool_ids = self.get_key_pools.map { |pool| pool['id'] }
-        pools = Katello::Pool.where(:cp_id => pool_ids)
+        key_pools = self.get_key_pools
+        pools = Katello::Pool.where(:cp_id => key_pools.map { |pool| pool['id'] })
         associations = Katello::PoolActivationKey.where(:activation_key_id => self.id)
         associations.map { |assoc| assoc.destroy! if pools.map(&:id).exclude?(assoc.pool_id) }
         pools.each do |pool|
-          Katello::PoolActivationKey.where(:pool_id => pool.id, :activation_key_id => self.id).first_or_create
+          pool_activation_key = Katello::PoolActivationKey.where(:pool_id => pool.id, :activation_key_id => self.id).first_or_create
+          quantity = key_pools.collect { |x| x["amount"] if x["id"] == pool.cp_id }.compact.first if key_pools.any?
```

Comment:
There is a lot of calls and logic going into this statement -- could we break it down a bit to reduce the complexity when reading it?